src/messages/Message.hpp

```diff
@@ -79,6 +90,8 @@ struct Message : boost::noncopyable {
     std::shared_ptr<MessageThread> replyThread;
     uint32_t count = 1;
     std::vector<std::unique_ptr<MessageElement>> elements;
+    std::vector<QString> liveUpdateEmoteNames;
+    EmoteProvider emoteProvider;
```

Comment:
Think on this a bit more to see if we can avoid storing random "live updates" code in `Message`